source/io/module_javaScriptInvoke.js

```diff
@@ -103,6 +102,61 @@
             return typeof value === 'number';
         };
 
+        var jsRefNumToJsObjectMap = new Map();
+        var jsObjectToJsRefNumMap = new Map();
+        var jsRefNumCookieCounter = 0;
+
+        var cacheRefNum = function (cookie, jsObject) {
+            jsRefNumToJsObjectMap.set(cookie, jsObject);
+            jsObjectToJsRefNumMap.set(jsObject, cookie);
```

Comment:
Do we need to consider caching only objects and not scalars?

Reply:
Yes! We discussed this in the past, but it seems I forgot to consider that case. Updated.